File: services/actuators/base-js/src/services/benchmark.ts
import { MongoClient, Collection, Document } from 'mongodb';

import { env } from '../env.js';

export class BenchmarkDataSaver {
  constructor(private readonly collection: Collection) {}

  async save(data: Document): Promise<void> {
    await this.collection.insertOne(data);
  }
}

export const mongoClient = new MongoClient(env.MONGO_URI);

export const database = mongoClient.db('arion_benchmark');

export const collection = database.collection('benchmark_data');

export const benchmarkDataSaver = new BenchmarkDataSaver(collection);

File: services/stream-processors/base/src/services/benchmark.py
import os

from pymongo import MongoClient


class BenchmarkDataSaver:
    def __init__(self, collection):
        self._collection = collection

    def save(self, benchmark_data):
        self._collection.insert_one(benchmark_data)


mongo_client = MongoClient(os.getenv("MONGO_URI"))
database = mongo_client.get_database("arion_benchmark")
collection = database.get_collection("benchmark_data")
benchmark_data_saver = BenchmarkDataSaver(collection)
